api/api.go

```diff
@@ -215,103 +215,113 @@ func NewServerRoutes(s Server) *mux.Router {
 	m.Handle(lib.AEIPFS.String(), s.Middleware(s.HandleIPFSPath))
-
-	proh := NewProfileHandlers(s.Instance, cfg.API.ReadOnly)
-	m.Handle(lib.AEMe.String(), s.Middleware(proh.ProfileHandler))
-	m.Handle(lib.AEProfile.String(), s.Middleware(proh.ProfileHandler))
-	m.Handle(lib.AEProfilePhoto.String(), s.Middleware(proh.ProfilePhotoHandler))
-	m.Handle(lib.AEProfilePoster.String(), s.Middleware(proh.PosterHandler))
-
-	m.Handle(lib.AEPeers.String(), s.Middleware(lib.NewHTTPRequestHandler(s.Instance, "peer.list"))).Methods(http.MethodPost)
-	m.Handle(lib.AEPeer.String(), s.Middleware(lib.NewHTTPRequestHandler(s.Instance, "peer.info"))).Methods(http.MethodPost)
-	m.Handle(lib.AEConnect.String(), s.Middleware(lib.NewHTTPRequestHandler(s.Instance, "peer.connect"))).Methods(http.MethodPost)
-	m.Handle(lib.AEDisconnect.String(), s.Middleware(lib.NewHTTPRequestHandler(s.Instance, "peer.disconnect"))).Methods(http.MethodPost)
-	m.Handle(lib.AEConnections.String(), s.Middleware(lib.NewHTTPRequestHandler(s.Instance, "peer.connections"))).Methods(http.MethodPost)
-	m.Handle(lib.AEConnectedQriProfiles.String(), s.Middleware(lib.NewHTTPRequestHandler(s.Instance, "peer.connectedqriprofiles"))).Methods(http.MethodPost)
-
-	if cfg.Remote != nil && cfg.Remote.Enabled {
-		log.Info("running in `remote` mode")
-
-		remh := NewRemoteHandlers(s.Instance)
-		m.Handle(lib.AERemoteDSync.String(), s.Middleware(remh.DsyncHandler))
-		m.Handle(lib.AERemoteLogSync.String(), s.Middleware(remh.LogsyncHandler))
-		m.Handle(lib.AERemoteRefs.String(), s.Middleware(remh.RefsHandler))
+	if !cfg.API.DisableWebui {
+		m.Handle(lib.AEWebUI.String(), s.Middleware(WebuiHandler))
 	}
 
-	dsh := NewDatasetHandlers(s.Instance, cfg.API.ReadOnly)
+	// aggregate endpoints
 	m.Handle(lib.AEList.String(), s.Middleware(lib.NewHTTPRequestHandler(s.Instance, "dataset.list"))).Methods(http.MethodPost, http.MethodGet)
 	m.Handle(lib.AEListRaw.String(), s.Middleware(lib.NewHTTPRequestHandler(s.Instance, "dataset.listrawrefs"))).Methods(http.MethodPost, http.MethodGet)
 	m.Handle(lib.AEPeerList.String(), s.Middleware(dsh.PeerListHandler(lib.AEPeerList.String())))
-	routeParams = newrefRouteParams(lib.AESave, false, false, http.MethodPost, http.MethodPut)
-	handleRefRoute(m, routeParams, s.Middleware(dsh.SaveHandler(lib.AESave.String())))
+	m.Handle(lib.AESQL.String(), s.Middleware(sqlh.QueryHandler))
+	m.Handle(lib.AEDiff.String(), s.Middleware(lib.NewHTTPRequestHandler(s.Instance, "dataset.diff"))).Methods(http.MethodPost, http.MethodGet)
+	m.Handle(lib.AEChanges.String(), s.Middleware(lib.NewHTTPRequestHandler(s.Instance, "dataset.changereport"))).Methods(http.MethodPost, http.MethodGet)
+
+	// access endpoints
+
+	// automation endpoints
+	m.Handle(lib.AEApply.String(), s.Middleware(tfh.ApplyHandler(lib.AEApply.NoTrailingSlash())))
+
+	// dataset endpoints
+	routeParams = newrefRouteParams(lib.AEWhatChanged, false, false, http.MethodGet, http.MethodPost)
```

Comment:
This should move `fsi` > `dataset`?

Reply:
Yes, and it is changing name from `fsi.whatchanged` to `dataset.componentstatus`.